pkg/cluster/delete.go

```diff
@@ -41,12 +48,68 @@ func Delete(
 			})
 		return err
 	}
+
 	clusterQuery := "arlon-cluster=" + name
 	apps, err := appIf.List(context.Background(),
 		&argoapp.ApplicationQuery{Selector: &clusterQuery})
 	if err != nil {
 		return fmt.Errorf("failed to list apps related to cluster: %s", err)
 	}
+
+	app, err := appIf.Get(context.Background(),
+		&apppkg.ApplicationQuery{
+			Name: &name,
+		})
+	if err != nil {
+		return fmt.Errorf("failed to get argocd application: %s", err)
+	}
+	typ := app.Labels["arlon-type"]
+	if typ == "cluster-app" {
+		log := logpkg.GetLogger()
+		repoUrl := app.Annotations[baseClusterRepoUrlAnnotation]
```

Comment:
repoUrl, repoRevision, and repoPath are not needed unless overridden is true, so move those inside of the if statement. And, as noted below, the whole thing should move to a separate function.